adafruit_rockblock.py

```diff
@@ -199,6 +209,187 @@ def model(self):
+        The response is in the form:
+        [<x>,<y>,<z>,<timestamp>]
+
+        <x>,<y>,<z> is a geolocation grid code from an earth centered Cartesian coordinate system,
+        using dimensions, x, y, and z, to specify location. The coordinate system is aligned
+        such that the z-axis is aligned with the north and south poles, leaving the x-axis
+        and y-axis to lie in the plane containing the equator. The axes are aligned such that
+        at 0 degrees latitude and 0 degrees longitude, both y and z are zero and
+        x is positive (x = +6376, representing the nominal earth radius in kilometres).
+        Each dimension of the geolocation grid code is displayed in decimal form using
+        units of kilometres. Each dimension of the geolocation grid code has a minimum value
+        of –6376, a maximum value of +6376, and a resolution of 4.
+        This geolocation coordinate system is known as ECEF (acronym earth-centered, earth-fixed),
+        also known as ECR (initialism for earth-centered rotational)
+
+        <timestamp> is a time_struct
+        The timestamp is assigned by the modem when the geolocation grid code received from
+        the network is stored to the modem's internal memory.
+        The timestamp used by the modem is Iridium system time, which is a running count of
+        90 millisecond intervals, since Sunday May 11, 2014, at 14:23:55 UTC.
+        The timestamp returned by the modem is a 32-bit integer displayed in hexadecimal form.
+        We convert the modem's timestamp and return it as a time_struct.
+        """
+        resp = self._uart_xfer("-MSGEO")
+        if resp[-1].strip().decode() == "OK":
+            temp = resp[1].strip().decode().split(":")[1].split(",")
+            ticks_since_epoch = int(temp[3], 16)
+            ms_since_epoch = (
+                ticks_since_epoch * 90
+            )  # convert iridium ticks to milliseconds
+
+            # milliseconds to seconds
+            # hack to divide by 1000 and avoid using limited floating point math which throws the
+            #    calculations off quite a bit, this should be accurate to 1 second or so
+            ms_str = str(ms_since_epoch)
+            substring = ms_str[0 : len(ms_str) - 3]
+            secs_since_epoch = int(substring)
+
+            # iridium epoch
+            iridium_epoch = time.struct_time(((2014), (5), 11, 14, 23, 55, 6, -1, -1))
+            iridium_epoch_unix = time.mktime(iridium_epoch)
+
+            # add timestamp's seconds to the iridium epoch
+            time_now_unix = iridium_epoch_unix + int(secs_since_epoch)
+
+            # convert to time struct
+            time_now = time.localtime(time_now_unix)
+
+            values = [
+                int(temp[0]),
+                int(temp[1]),
+                int(temp[2]),
+                time_now,
+            ]
+            return values
+        return None
+
+    @property
+    def timestamp(self):
```

Comment:
Change name to `system_time`?
![image](https://user-images.githubusercontent.com/8755041/104068452-cde3f380-51b9-11eb-99a8-8a4db8725b42.png)